runtime/lua/vim/lsp/semantic_tokens.lua

```diff
@@ -575,6 +585,76 @@ function M.stop(bufnr, client_id)
   end
 end
 
+--- Return the semantic token(s) at the given position.
+--- If called without argument, returns the token under the cursor.
+---
+---@param bufnr number|nil Buffer number (0 for current buffer, default)
+---@param row number|nil Position row (default cursor position)
+---@param col number|nil Position column (default cursor position)
+---
+---@return table[]|nil tokens Table of tokens at position
+function M.get_at_pos(bufnr, row, col)
+  if bufnr == nil or bufnr == 0 then
+    bufnr = api.nvim_get_current_buf()
+  end
+
+  local highlighter = STHighlighter.active[bufnr]
+  if not highlighter then
+    return
+  end
+
+  if not (row and col) then
+    local cursor = api.nvim_win_get_cursor(0)
+    row = cursor[1] - 1
+    col = cursor[2]
+  end
+
+  local tokens = {}
+  for _, client in pairs(highlighter.client_state) do
+    local highlights = client.current_result.highlights
+    if highlights then
+      local idx = binary_search(highlights, row)
+      for i = idx, #highlights do
+        local token = highlights[i]
+
+        if token.line > row then
+          break
+        end
+
+        if token.start_col <= col and token.end_col > col then
+          tokens[#tokens + 1] = token
+        end
+      end
+    end
+  end
+  return tokens
+end
+
+--- Show the semantic token(s) under the cursor
+function M.show_at_cursor()
+  local tokens = M.get_at_pos()
+  if not tokens then
+    return
+  end
```

Comment:
```suggestion
--- Show the semantic token(s) under the cursor
---@param tokens table[]|nil tokens to print, defaults to tokens at cursor
function M.show(tokens)
  tokens = tokens or M.get_at_pos()
  if not tokens then
    return
  end
```

This is what I had in mind - it would allow to combine it with `M.get_at_pos` to print the tokens from another position.
But I don't really have a concrete use-case for it.